oper8/rollout_manager.py

```diff
@@ -306,6 +320,27 @@ def rollout(  # pylint: disable=too-many-locals,too-many-statements
 
         phase2_complete = phase1_complete
         phase2_exception = None
+
+        # After unsuccessful deploy.
+        if (not phase1_complete or phase1_failed) and self._after_deploy_unsuccessful:
+            log.debug2("Running after-deploy-unsuccessful")
+            try:
+                is_after_deploy_unsuccessful_completed = (
+                    self._after_deploy_unsuccessful(self._session)
```

Comment:
If we go with the addition of a failure flag:

```suggestion
                    self._after_deploy_unsuccessful(self._session, phase1_failed)
```